src/base.js

```diff
@@ -0,0 +1,24 @@
+ class Base {
+   constructor(elt){
+     this.elt = elt;
+     const closeElt = document.createElement('div');
+     closeElt.classList.add('close');
+     closeElt.innerText = 'X';
+     if (this.elt.classList.contains('toast')){
+       this.elt.appendChild(closeElt);
+     } else {
+       const modalElt = this.elt.querySelector('.modal_content');
```

Comment:
I might have missed it in ES5 but you should not use any mentions of children in parent classes. Please, remove it from your logic. Parent class shouldn't be aware of by whom he will be extended